### ISSUE TITLE
How to document writing system used in a treebank?

### ISSUE THREAD

**Issue by Abhishek-P** [UniversalDependencies/docs#1032]
This is a case I came across when using UD Sanskrit (https://universaldependencies.org/sa/index.html) treebank(s). 

The two treebanks use two different scripts, [UFAL](https://universaldependencies.org/treebanks/sa_ufal/index.html) uses [devanagari](https://scriptsource.org/cms/scripts/page.php?item_id=script_detail&key=Deva) while [Vedic](https://universaldependencies.org/treebanks/sa_vedic/index.html) uses latin. 

I suspect this maybe true for some other languages (yet to do an audit) (I am also assuming other such cases, script mixing is not a case we have to worry about with this)

Currently, the tree bank page does not provide any explicit information about the script - although this can be inferred from the examples in the morphology overview section.

I think it would be nice to have that information surfaced more clearly in the treebank page since this can be an important tree bank characteristic to keep in mind for certain uses. 

My preliminary proposal for it would be to add it as part of the description similar to Genre with a hyperlink to a scholarly source on scripts (scriptsource?)

> License: CC BY-SA 4.0
> 
> Genre: fiction
> Script: devanagari

**Comment by dan-zeman** [comment]
[ISO 15924](https://en.wikipedia.org/wiki/ISO_15924) provides codes suitable for such a metadata item. There are probably finer distinctions that could be made about the spelling rules in the treebank, but those would be difficult to capture systematically, and ISO codes of scripts would be an improvement over no info (current status).

**Comment by Abhishek-P** [comment]
I can make a change in the `sa` treebanks pages using the ISO 15924 for a start. 
Let me know if and where this needs to be discussed and documented (templatized) for future work.

**Comment by Abhishek-P** [comment]
I did a quick check of the treebank comparison pages (those linked in home page) `sa` is the only case I caught of treebanks having different scripts.

**Comment by dan-zeman** [comment]
> I did a quick check of the treebank comparison pages (those linked in home page) `sa` is the only case I caught of treebanks having different scripts.

This is true at the moment as far as I know, but there are other languages that could use multiple writing systems, so it is definitely a property of the treebank rather than the language.

**Comment by dan-zeman** [comment]
> I can make a change in the `sa` treebanks pages using the ISO 15924 for a start. Let me know if and where this needs to be discussed and documented (templatized) for future work.

I will raise this at a future meeting of the core group. Assuming there won't be objections, these are the next steps:

- Document it next to other metadata in the Release checklist.
- Add it to validation infrastructure (the line will be required and must contain valid values, e.g. `Script: Latn` for Latin-based alphabets).
- Announce it to the mailing list and make sure that all treebanks have the new line in their READMEs. Also add it to the template used when creating new repositories.
- Make sure that the script that generates treebank hub pages (at release time) copies this information to the pages.

**Comment by amir-zeldes** [comment]
> This is true at the moment as far as I know, but there are other languages that could use multiple writing systems, so it is definitely a property of the treebank rather than the language.

Another candidate is UD_Egyptian, which uses Schenkel transcription rather than hieroglyphs or Gardiner codes, either of which would be conceivable for Egyptian.

**Comment by robvanderg** [comment]
If it should be automated, it can be tricky to find code for this (as script is ambiguous when searching for code), here are some existing solutions (last one by me, optimized for speed not RAM):

https://github.com/cisnlp/GlotScript
https://robvanderg.github.io/scripts/scripts/

**Comment by dan-zeman** [comment]
As a first step, the newly generated [table of UD languages](https://universaldependencies.org/languages.html) shows one or more scripts (ISO 15924 codes) for each language. I focused on scripts that are currently used in UD data, so, for example, I did not list Cyrillic for Serbian because we have only one treebank and it is encoded in Latin. More scripts can be added in the future as needed.

The treebank-level metadata could be the next step and I suppose then the validator could check both that the treebank README lists a script we expect in that language, and that the majority of the Unicode characters in the data actually belong to that script.

**Comment by bguil** [comment]
Note that since 2.18, there are at least one treebank with a mix of writing system.
In UD_Brahui-Kholum, there are 52 sentences in a Latin-based script and 12 in right-to-left Arabic-based script:  https://universal.grew.fr/?custom=6a1ac9283d477
So we may need to have this metadata at the sentence level.

**Comment by dan-zeman** [comment]
> So we may need to have this metadata at the sentence level.

It depends on what purpose it would serve. We could even consider word level or character level because it is not unusual for a treebank to occasionally contain foreign characters, especially when the source is Wikipedia. I never thought of annotating these—if someone wants it, they can simply [query the Unicode database](https://github.com/UniversalDependencies/tools/blob/8a50b92bddd78e3a29b639bcbcd4a05dc23de4c1/survey_scripts.pl#L221-L226) for individual characters. The Brahui treebank is unusual because it really combines full sentences from two different sources in two different scripts that the language uses, but it could still be processed the same way and maybe we do not need to add sentence-level `script = Latn` to all sentences of all other treebanks.

On second thought, I'm starting to doubt my original proposal of registering the script in treebank's README. It would be another thing that can be extracted from the data; doubling it in the README only means that we must extract it automatically anyway and check that the treebank maintainer entered it in the README correctly. So maybe it is enough if we survey the scripts automatically at release time, just as we do with a bunch of other things, and publish it in the documentation on the web. I have implemented a first version of that, the results [can be viewed here](https://universaldependencies.org/survey-scripts.html).

**Comment by Stormur** [comment]
I commented in the other issue, but in fact this can be done extracting the Unicode codes for scripts (very basic actually, e.g. GREEK, ETHIOPIC, KIRAT RAI...), and possibly a solution is to give their percentages over the total. So spurious characters do not appear at the same level as the main script, which will be above 90%. I think it is not so uncommon.

**Comment by Stormur** [comment]
Oops, sorry, I did not realise that some ISO codes are used. But my suggestion of giving the composition by % remains!

**Comment by dan-zeman** [comment]
Done in b0069be.
